File: CSharpPractice/LeetcodePractice/Leetcode2696.cs
public class Solution2696 {
    public int MinLength(string s) 
    {
        Stack<char> theStack = new Stack<char>();
        for(int i=0;i<s.Length;i++)
        {
            if(theStack.Count>0)
            {
                char lastElement = theStack.Peek();
                if((s[i]=='B'&&lastElement=='A')||(s[i]=='D'&&lastElement=='C'))
                {
                    theStack.Pop();
                }
                else
                {
                    theStack.Push(s[i]);
                }

            }
            else
            {
                theStack.Push(s[i]);
            }
        }
        return theStack.Count;
        
    }
}

File: JavaScriptPractice/Leetcode2696.js
var minLength = function(s) 
{
    let n = s.length;
    let theStack = []
    for(let i=0;i<n;i++)
    {
        if((s[i]=='B'&&theStack[theStack.length-1]=='A')||(s[i]=='D'  &&theStack[theStack.length-1]=='C'))
        {
            theStack.pop();
        }
        else
        {
            theStack.push(s[i]);
        }
    }

    return theStack.length;
    
};